### PR TITLE
Added first Go subcommand: `job list`.

### PR DESCRIPTION
**Pull request by rgo3**
This subcommand is currently not user-facing, it will be once
all job subcommands will be refactored using Golang.

Co-authored-by: Robin Gögge <r.goegge@outlook.com>
Co-authored-by: Bilal Amarni <bilal.amarni@gmail.com>
Co-authored-by: Armand Grillet <armand.grillet@outlook.com>

### CODE REVIEW

File: pkg/cmd/core.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import (
+	"github.com/dcos/dcos-cli/api"
+	"github.com/dcos/dcos-core-cli/pkg/cmd/job"
+	"github.com/spf13/cobra"
+)
+
+// NewCORECommand creates the `core` command with all the available subcommands.
+func NewCORECommand(ctx api.Context) *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "core",
```
Comment (bamarni) [inline pkg/cmd/core.go]: should be `dcos`

File: pkg/cmd/core.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package cmd
+
+import (
+	"github.com/dcos/dcos-cli/api"
+	"github.com/dcos/dcos-core-cli/pkg/cmd/job"
+	"github.com/spf13/cobra"
+)
+
+// NewCORECommand creates the `core` command with all the available subcommands.
+func NewCORECommand(ctx api.Context) *cobra.Command {
```
Comment (bamarni) [inline pkg/cmd/core.go]: The case should be `NewCoreCommand`, however there is no `core` command (see my comment below). I think something like `NewCommand`, `NewRootCommand` or `NewDCOSCommand` would be more suited .

File: pkg/metronome/client.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	"encoding/json"
+
+	"github.com/dcos/dcos-cli/pkg/httpclient"
+)
+
+
+// Constants for embed query parameters
+const embedActiveRuns = "activeRuns"
+const embedSchedules = "schedules"
+const embedHistory = "history"
```
Comment (bamarni) [inline pkg/metronome/client.go]: This is unexported and unused.

File: pkg/metronome/client.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+// Jobs returns a list of all job definitions.
+func (c *Client) Jobs() ([]Job, error) {
+
+	req, err := c.http.NewRequest("GET","/v1/jobs", nil, httpclient.FailOnErrStatus(true))
+	if err != nil {
+		return nil, err
+	}
+
+	// Add embed query parameters to the request URL
+	embedForJobsHistory := []string{embedActiveRuns, embedSchedules, embedHistorySummary}
```
Comment (bamarni) [inline pkg/metronome/client.go]: Should this be configurable? Eg. accepting a struct as a parameter indicating what embed fields are needed.

File: pkg/metronome/types.go [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+	ID          string            `json:"id"`
+	Description string            `json:"description"`
+	Labels      map[string]string `json:"labels"`
+	Run         run               `json:"run"`
+
+	// these properties depend on the embed parameters when querying the /v1/jobs endpoints
+	ActiveRuns     []activeRun        `json:"activeRuns,omitempty"`
+	HistorySummary *jobHistorySummary `json:"historySummary,omitempty"`
+	History        *jobHistory        `json:"history,omitempty"`
+	Schedules      []schedule         `json:"schedules,omitempty"`
```
Comment (bamarni) [inline pkg/metronome/types.go]: I think the nested types in this struct should be exported, they will be needed for other endpoints (`/v1/jobs/{jobId}/runs` or `/v1/jobs/{jobId}/schedules`), it also lets users typehinting them if they want to process them.

File: pkg/metronome/types.go [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+	ID                      string `json:"id"`
+	Cron                    string `json:"cron"`
+	TimeZone                string `json:"timeZone"`
+	StartingDeadlineSeconds int    `json:"startingDeadlineSeconds"`
+	ConcurrencyPolicy       string `json:"concurrencyPolicy"`
+	Enabled                 bool   `json:"enabled"`
+	NextRunAt               string `json:"nextRunAt"`
+}
+
+// Status ...
```
Comment (bamarni) [inline pkg/metronome/types.go]: You can trick golint but I don't fall for this :smile:

File: pkg/metronome/types.go [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+func (j *Job) Status() string {
+	switch {
+	case j.ActiveRuns != nil:
+		return "Running"
+	case j.Schedules == nil, len(j.Schedules) == 0:
```
Comment (bamarni) [inline pkg/metronome/types.go]: checking the length is enough (see https://tour.golang.org/moretypes/12)

File: pkg/metronome/client.go (hunk 3 above)
Comment (rgo3) [reply pkg/metronome/client.go]: True, but this will be used in one of the other methods in this file.

File: pkg/cmd/job/job.go [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+	baseURL, _ := cluster.Config().Get("job.url").(string)
+	if baseURL == "" {
+		baseURL = cluster.URL() + "/service/metronome"
+	}
+
+	return metronome.NewClient(httpclient.New(
+		baseURL,
+		httpclient.Logger(ctx.Logger()),
+		httpclient.ACSToken(cluster.ACSToken()),
```
Comment (bamarni) [inline pkg/cmd/job/job.go]: It'd be good to have a timeout here, `httpclient.Timeout(cluster.Timeout())`.

**Pull request merged**